modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2SynapseEnvironment.java

```diff
@@ -235,21 +236,27 @@ private void setSeverUnitTestMode(ServerContextInformation contextInformation) {
             //get unit test received port
             String requestPort = System.getenv("UNIT_TEST_PORT");
 
-            //set default port for TCP server connection
-            int unitTestingAgentPort = 9007;
-            if (requestPort != null && !requestPort.isEmpty()) {
-                try {
+            //check port for TCP server connection
+            int unitTestingAgentPort;
+            try {
+                if (requestPort != null && !requestPort.isEmpty()) {
                     unitTestingAgentPort = Integer.parseInt(requestPort);
-                } catch (NumberFormatException e) {
-                    log.error("Given port is not in valid format", e);
+
+                    if (unitTestingAgentPort > 1023) {
+                        UnitTestingExecutor testExecutor = UnitTestingExecutor.getExecuteInstance();
+                        testExecutor.setServerPort(unitTestingAgentPort);
+                        testExecutor.setSynapseConfiguration(this.synapseConfig);
+                        testExecutor.start();
+                    } else {
+                        log.error("Input a TCP port higher than 1023 to start the unit testing framework");
+                    }
+                } else {
+                    log.error("Given port can not be empty or null, failed to start unit testing framework");
                 }
-            }
 
-            //execute synapse unit testing executor
-            UnitTestingExecutor testExecutor = UnitTestingExecutor.getExecuteInstance();
-            testExecutor.setServerPort(unitTestingAgentPort);
-            testExecutor.setSynapseConfiguration(this.synapseConfig);
-            testExecutor.start();
+            } catch (NumberFormatException e) {
+                log.error("Given port is not in valid format", e);
```

Comment:
Shall we print what tried to parse?

Reply:
Also better if we could append the error with "failed to start unit testing framework" since it won't start thereafter.

Reply:
Done